src/UI/Art/SpriteArtist/PlayerSetupColorFactionArtist.java

```diff
@@ -84,6 +92,12 @@ public static void draw(Graphics g, IController controller)
     BufferedImage image = SpriteLibrary.createTransparentSprite(myWidth, myHeight);
     Graphics myG = image.getGraphics();
 
+    if( !donePreloading )
+    {
+      isp.join();
+      isp = null;
+    }
```

Comment:
If I'm reading this right:
If we aren't done pre-loading before this gets here, the game just freezes until we are.
If we *are* done pre-loading, then isp never becomes null.

Is that accurate?

Reply:
Mmmm, yes. Not exactly what was intended though.